lua/goto-preview/lib.lua

```diff
@@ -112,26 +116,13 @@ local function set_title_pos()
   return M.conf.preview_window_title.enable and M.conf.preview_window_title.position or nil
 end
 
-M.open_floating_win = function(target, position, opts)
-  local buffer = type(target) == "string" and vim.uri_to_bufnr(target) or target
-  local bufpos = { vim.fn.line "." - 1, vim.fn.col "." } -- FOR relative='win'
-  local zindex = vim.tbl_isempty(M.windows) and 1 or #M.windows + 1
-
-  opts = opts or {}
+local function create_preview_win(buffer, bufpos, zindex, opts)
   local enter = function()
-    if opts.focus_on_open ~= nil then
-      return opts.focus_on_open
-    else
-      return M.conf.focus_on_open
-    end
+    return opts.focus_on_open or M.conf.focus_on_open
```

Comment:
This simplification doesn't achieve the desired outcome. With this the result of `enter()` can  be `nil` when it should only ever be `true` or `false`. `vim.api.nvim_open_win`'s second argument takes a boolean, not a boolean or nil.

However the following should do the trick:
```return opts.focus_on_open or M.conf.focus_on_open or false```

Reply:
I see, you're right since this change would break the current flow users expect. I'll add an option to turn this feature on in the configs.

Also, nice catch with focus_on_open, I updated it.